### DISCUSSION TITLE
Настройка внешнего редактора far2l

### DISCUSSION THREAD

**Discussion by Mischutka** [elfmz/far2l#1160] [Q&A]
Народ, подскажите пожалуйста, как настроить комбинацию клавиш Alt+F4, чтобы при её нажатии запускался внешний редактор far2l, а не закрывался сам файловый менеджер?

Операционная система: Astra Linux (Orel)

![far2l_editor2](https://user-images.githubusercontent.com/93511177/147684020-2a6571de-c0bc-4acb-b339-d122a610e86a.png)

**Comment by elfmz** [comment]
Alt-F4 глобальный хоткей зарезервированный операционкой, его поменять то можно (в настройках десктопного шелла), но это не оч хорошая идея, так как все к нему привыкли)
Если на клавиатуре есть кнопка Win - то можно воспользоваться комбинацией Win+F4 т.к. far2l интерпретирует Win так же как Alt. Если конечно Win+F4 тоже не занята чемнить другим, но в таком случае это другое можно перенастроить т..к. этот хоткей гораздо более специфичен.

**Comment by Mischutka** [comment]
> Если на клавиатуре есть кнопка Win - то можно воспользоваться комбинацией Win+F4 т.к. far2l интерпретирует Win так же как Alt.

Попробовал — вообще никакой реакции.

**Comment by elfmz** [comment]
Именно Win+F4 не обрабатывается как запуск альт редактора или вообще Win+ не срабатывают как полагается? Например Win+F7 должен запускать поиск файлов, так же как Alt+F7 и т.п.
Кстати опять же вопрос - какую версию вы используете? TTY или GUI? В TTY фокус с Win не работает (до последних изменений с использованием Xi для ввода)

**Reply by Mischutka** [reply]
Клавиша Win срабатывает — появляется меню "Пуск". Не срабатывает именно Win+F4 как вызов внешнего редактора.

**Reply by Mischutka** [reply]
> Кстати опять же вопрос - какую версию вы используете? TTY или GUI?

Затрудняюсь с ответом. Как это можно определить?

**Reply by elfmz** [reply]
ну если вы far2l запускаете в терминале и в терминале же отображается его интерфейс - то это TTY версия, а если же запускается в отдельном (своем) окне - то гуишная.

**Reply by Mischutka** [reply]
> ну если вы far2l запускаете в терминале и в терминале же отображается его интерфейс - то это TTY версия, а если же запускается в отдельном (своем) окне - то гуишная.

В отдельном окне (даже если запускать через ввод во fly-терминале команды far2l).

**Reply by elfmz** [reply]
Ну значит GUI, скажите тогда какой у вас desktop shell, включая версию. Сразу скажу что если у вас гном, то там левый вин занят шеллом, можно установить gnome-tweaks и переназначить шелл на правый вин, или попробовать правый Win+F4 (тут обещать не могу так как у меня самого клава только с одной кнопкой вин.)

**Reply by Mischutka** [reply]
> Ну значит GUI, скажите тогда какой у вас desktop shell, включая версию. Сразу скажу что если у вас гном, то там левый вин занят шеллом, можно установить gnome-tweaks и переназначить шелл на правый вин, или попробовать правый Win+F4 (тут обещать не могу так как у меня самого клава только с одной кнопкой вин.)

fly-wm version 2.40.3.

**Comment by unxed** [comment]
хм, у меня в wx Win+F7 работает как просто F7. но может это специфика ноутбучной клавиатуры
